extensions/lifecycle/tasks/LifecycleDeleteObjectTask.js

```diff
@@ -170,6 +177,27 @@ class LifecycleDeleteObjectTask extends BackbeatTask {
         });
     }
 
+    /**
+     * Throws error if object has a 'PENDING' replication status
+     * @param {ActionQueueEntry} entry entry object
+     * @param {Logger} log logger instance
+     * @param {Function} done callback
+     * @returns {undefined}
+     */
+    _checkReplicationStatus(entry, log, done) {
+        return this._getMetadata(entry, log, (err, objMD) => {
+            if (err) {
+                return done(err);
+            }
+            const replicationStatus = objMD.getReplicationStatus();
+            if (replicationStatus === 'PENDING') {
```

Comment:
What about 'PROCESSING' and 'FAILED' status?
Should we check `replicationStatus && replicationStatus !== 'COMPLETED'` ?

Replication status:
- PROCESSING: 
      - (multiple sites) one site is replication COMPLETED while other sites are still PENDING
      - entry had a FAILED replication and was picked up by "FailedKafkaEntries" cronjob that updated the global replication status to PROCESSING along with sending an entry to the replication topic so that the queue processor re-attempts replication.
- PENDING: 
      - Fresh new object waiting to be replicated. PENDING is set by Cloudserver for objects created in a bucket with replication enabled. Meaning that no replication has been attempted before on the specific object.
- FAILED:
      - at least one replication site has FAILED

Reply:
Good catch, i think we should also include these other statuses